packages/medusa/src/api/routes/store/product-types/list-product-types.ts

```diff
@@ -0,0 +1,198 @@
+ *            format: date
+ *   - in: query
+ *     name: updated_at
+ *     description: Date comparison for when resulting product types were updated.
+ *     schema:
+ *       type: object
+ *       properties:
+ *         lt:
+ *            type: string
+ *            description: filter by dates less than this date
+ *            format: date
+ *         gt:
+ *            type: string
+ *            description: filter by dates greater than this date
+ *            format: date
+ *         lte:
+ *            type: string
+ *            description: filter by dates less than or equal to this date
+ *            format: date
+ *         gte:
+ *            type: string
+ *            description: filter by dates greater than or equal to this date
+ *            format: date
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.store.productTypes.list()
+ *       .then(({ product_types }) => {
+ *         console.log(product_types.length);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request GET 'https://medusa-url.com/store/product-types' \
+ *       --header 'Authorization: Bearer {api_token}'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Product Type
+ * responses:
+ *  "200":
+ *    description: OK
+ *    content:
+ *      application/json:
+ *        schema:
+ *          properties:
+ *            product_types:
+ *              $ref: "#/components/schemas/product_type"
+ *            count:
+ *              type: integer
+ *              description: The total number of items available
+ *            offset:
+ *              type: integer
+ *              description: The number of items skipped before these items
+ *            limit:
+ *              type: integer
+ *              description: The number of items per page
+ *  "400":
+ *    $ref: "#/components/responses/400_error"
+ *  "401":
+ *    $ref: "#/components/responses/unauthorized"
+ *  "404":
+ *    $ref: "#/components/responses/not_found_error"
+ *  "409":
+ *    $ref: "#/components/responses/invalid_state_error"
+ *  "422":
+ *    $ref: "#/components/responses/invalid_request_error"
+ *  "500":
+ *    $ref: "#/components/responses/500_error"
+ */
+export default async (req, res) => {
+  const typeService: ProductTypeService =
+    req.scope.resolve("productTypeService")
+
+  const { listConfig, filterableFields } = req
+  const { skip, take } = req.listConfig
+
+  const [types, count] = await typeService.listAndCount(
+    filterableFields,
+    listConfig
+  )
+
+  res.status(200).json({
+    product_types: types,
+    count,
+    offset: skip,
+    limit: take,
+  })
+}
+
+export class StoreGetProductTypesPaginationParams {
```

Comment:
Nice, I think all remove the definition StoreGetProductTypesPaginationParams and use FindPaginationParams directly on StoreGetProductTypesParams.
`export class StoreGetProductTypesParams extends FindPaginationParams`

The only side effect will be that the default limit will be 20 instead of 10 which I don't expect to be an issue.

I'll apply the same the equivalent Admin endpoint.

Reply:
Done, see 0ee84569846c46b5fa54227249dbbafd10b87715

Reply:
yeah I don't think the default limit is a problem as it will be consistent when everything will extend it :)